framework/src/modules/token/method.ts

```diff
@@ -706,9 +744,57 @@ export class TokenMethod extends BaseMethod {
 		this.events.get(TokenIDSupportedEvent).log(methodContext, tokenID);
 	}
 
-	public async removeSupportTokenID(methodContext: MethodContext, tokenID: Buffer): Promise<void> {
-		await this.stores.get(SupportedTokensStore).removeSupportForToken(methodContext, tokenID);
-		this.events.get(TokenIDSupportRemovedEvent).log(methodContext, tokenID);
+	public async removeSupport(methodContext: MethodContext, tokenID: Buffer): Promise<void> {
+		const [chainID] = splitTokenID(tokenID);
+
+		const allTokensSupported = await this.stores
+			.get(SupportedTokensStore)
+			.has(methodContext, ALL_SUPPORTED_TOKENS_KEY);
+
+		if (allTokensSupported) {
+			throw new Error('All tokens are supported.');
+		}
+
+		if (tokenID.equals(this.getMainchainTokenID()) || chainID.equals(this._config.ownChainID)) {
+			throw new Error('Cannot remove support for the specified token.');
+		}
+
+		const isChainSupported = await this.stores
+			.get(SupportedTokensStore)
+			.has(methodContext, chainID);
+
+		if (!isChainSupported) {
+			this.events.get(TokenIDSupportRemovedEvent).log(methodContext, tokenID);
+			return;
+		}
+
+		try {
+			const supportedTokens = await this.stores
+				.get(SupportedTokensStore)
+				.get(methodContext, chainID);
+
+			if (supportedTokens) {
```

Comment:
there is no case where supportedTokens return `undefined` or `null` so this `if` is not necessary